api/numaresourcesoperator/v1/helper/nodegroup/nodegroup.go

```diff
@@ -49,9 +49,22 @@ func FindTrees(mcps *mcov1.MachineConfigPoolList, nodeGroups []nropv1.NodeGroup)
 	for idx := range nodeGroups {
 		nodeGroup := &nodeGroups[idx] // shortcut
 
-		if nodeGroup.MachineConfigPoolSelector == nil {
+		if nodeGroup.MachineConfigPoolSelector == nil && nodeGroup.NodeSelector == nil {
+			klog.Errorf("node group selectors are empty, must specify one selector: %+v", nodeGroup)
 			continue
 		}
+		if nodeGroup.MachineConfigPoolSelector != nil && nodeGroup.NodeSelector != nil {
+			klog.Errorf("only one node group selector is allowed: %+v", nodeGroup)
+			continue
+		}
+
+		if nodeGroup.NodeSelector != nil {
+			result = append(result, Tree{
+				NodeGroup: nodeGroup,
+			})
+			continue
+		}
```

Comment:
this is good, but seems misplaced. I'd move (a slightly amended version of) this logic in https://github.com/openshift-kni/numaresources-operator/blob/main/pkg/validation/validation.go .

The validation logic is called much earlier in each reconciliation step. Feel free to add this logic into an existing function, or to create a new one.

Reply:
removed the reported errors but still kept the conditions